04-request-lifecycle/01-nestjs-components/filters/http-error.filter.ts

```diff
@@ -1,5 +1,25 @@
-import { ArgumentsHost, ExceptionFilter } from "@nestjs/common";
+import { ArgumentsHost, Catch, ExceptionFilter, HttpException } from "@nestjs/common";
+import { Request, Response } from 'express';
+import {appendFileSync} from 'fs';
 
+@Catch(HttpException)
 export class HttpErrorFilter implements ExceptionFilter {
-  catch(exception: any, host: ArgumentsHost) {}
+  catch(exception:  HttpException, host: ArgumentsHost) {
+    const ctx = host.switchToHttp();
+    const response = ctx.getResponse<Response>();
+    const status = exception.getStatus();
+    const message = exception.message;
+    console.log(exception);
+    const timestamp = new Date().toISOString();
+
+    appendFileSync('./errors.logs', `[${timestamp}] ${status} - ${message}`);
```

Comment:
И последнее - по условию мы должны писать логи в errors.log файл, и добавлять в конце каждой строки перенос, то есть:
```suggestion
    appendFileSync('errors.log', `[${timestamp}] ${status} - ${message}\n`);
```

Reply:
Спасибо, поправил